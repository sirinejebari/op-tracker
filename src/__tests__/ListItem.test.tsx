import React from 'react';
import ReactDOM from 'react-dom';
import ListItem from '../components/list-item/list-item';
import renderer from 'react-test-renderer';

const item = { date: "2019-05-20T17:59:56+02:00", "reference": "BBCGIYECO7LB", "operator": "Bigblue System", "subtype": "status_update", "short": "DELIVERY_EXCEPTION", "description": "Transmitted to the warehouse" }

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ListItem firstItem={true} rowType='light' key={item.reference} order={item} />, div);
  ReactDOM.unmountComponentAtNode(div);
})

test('render with the right classes: exception', () => {
  const component = renderer.create(
    <ListItem firstItem={true} rowType='light' key={item.reference} order={item} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree.props.className).toContain('animate')
  expect(tree.props.className).toContain('exception')
});

test('render with the right classes: delivered', () => {
  const item = { date: "2019-05-20T17:59:56+02:00", "reference": "BBCGIYECO7LB", "operator": "Bigblue System", "subtype": "status_update", "short": "DELIVERED", "description": "Transmitted to the warehouse" }

  const component = renderer.create(
    <ListItem firstItem={true} rowType='light' key={item.reference} order={item} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree.props.className).toContain('animate')
  expect(tree.props.className).toContain('delivered')
})

