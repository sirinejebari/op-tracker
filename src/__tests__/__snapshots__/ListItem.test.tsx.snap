// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with the right classes: delivered 1`] = `
<div
  className="item animate   delivered light"
>
  <div
    className="reference"
  >
    BBCGIYECO7LB
  </div>
  <div
    className="operator"
  >
    Bigblue System
  </div>
  <div
    className="status"
  >
    <span>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-check-circle fa-w-16 "
        data-icon="check-circle"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
       Delivered
    </span>
  </div>
  <div
    className="description"
  >
    Transmitted to the warehouse
  </div>
  <div
    className="date-updated"
  >
    20/05/2019 5:59:56 pm
  </div>
</div>
`;

exports[`render with the right classes: exception 1`] = `
<div
  className="item animate  exception  light"
>
  <div
    className="reference"
  >
    BBCGIYECO7LB
  </div>
  <div
    className="operator"
  >
    Bigblue System
  </div>
  <div
    className="status"
  >
    <span>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-exclamation-circle fa-w-16 "
        data-icon="exclamation-circle"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
       Delivery Exception
    </span>
  </div>
  <div
    className="description"
  >
    Transmitted to the warehouse
  </div>
  <div
    className="date-updated"
  >
    20/05/2019 5:59:56 pm
  </div>
</div>
`;
